### PR TITLE
Pull request for #77: Replace <*> with <source file name> in <sconscript_*.log>

### PR DESCRIPTION
**Pull request by stanfordquan**
@arosenbe I added a step in builders for the user to specify a bit in the sconscript_*.log as mentioned in the title. The user specify an argument in `env.BuildSomething` as `log_ext = *`. It should be fairly obvious what happens. There's a PR in `template` as a companion to this.

I would appreciate it if you can help out updating the unittests since you're the Mock expert.

Thanks!

### CODE REVIEW

**Comment by arosenbe** [comment]
#82 has some fixes for the tests of build_table. We'll hold off here until that goes through to avoid redundant work.

**Comment by stanfordquan** [comment]
Tests run fully now, @arosenbe. Thanks!

File: gslab_scons/builders/build_lyx.py [hunk 1]
```diff
@@ -39,7 +39,11 @@ def build_lyx(target, source, env):
 
     misc.check_code_extension(source_file, 'lyx')
     newpdf   = source_file.replace('.lyx','.pdf')
-    log_file = target_dir + '/sconscript.log'
+    try:
+        log_ext = '_%s' % env['log_ext']
+    except:
```
Comment (stanfordquan) [inline gslab_scons/builders/build_lyx.py]: Sure, but that broke a bunch of unittests that feed weird objects as `Env`

File: gslab_scons/builders/build_lyx.py [hunk 2]
```diff
@@ -39,7 +39,11 @@ def build_lyx(target, source, env):
...
+        log_ext = ''
+    log_file    = target_dir + ('/sconscript%s.log' % log_ext)
```
Comment (stanfordquan) [inline gslab_scons/builders/build_lyx.py]: Done

**Comment by stanfordquan** [comment]
@arosenbe I responded to tyour PR but that broke a number of unittests that feed weird objects as `Env`, so `env['log_ext']` raises different kinds of errors instead of KeyError and break the code.

**Comment by stanfordquan** [comment]
@arosenbe I fixed the unittests.

**Pull request merged**